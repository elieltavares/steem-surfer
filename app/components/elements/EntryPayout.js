import React, { Component } from 'react';

import PropTypes from 'prop-types';

import { Popover } from 'antd';

import EntryPayoutContent from './EntryPayoutContent';

class EntryPayout extends Component {
  render() {
    const { children } = this.props;

    return (
      <Popover
        key="payout-popover"
        content={<EntryPayoutContent {...this.props} />}
        placement="bottom"
      >
        {children}
      </Popover>
    );
  }
}

EntryPayout.propTypes = {
  children: PropTypes.element.isRequired
};

export default EntryPayout;
