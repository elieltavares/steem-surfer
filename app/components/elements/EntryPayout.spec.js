/* eslint-disable */
import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import EntryPayout from './EntryPayout';

Enzyme.configure({ adapter: new Adapter() });

describe('Payout info component', () => {
  it('should match exact snapshot', () => {
    const props = {
      global: {
        currencyRate: 1,
        currencySymbol: '$'
      },
      entry: {
        pending_payout_value: '430.124 SBD',
        promoted: '0.000 SBD',
        total_payout_value: '0.000 SBD',
        curator_payout_value: '0.000 SBD',
        last_payout: '1970-01-01T00:00:00',
        cashout_time: '2018-09-18T12:30:42'
      }
    };

    const component = shallow(
      <EntryPayout {...props}>
        <a className="post-total">$ 430.12</a>
      </EntryPayout>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
