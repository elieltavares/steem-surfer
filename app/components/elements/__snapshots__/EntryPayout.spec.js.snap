// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payout info component should match exact snapshot 1`] = `
<a
  className="post-total"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  $ 430.12
</a>
`;
